### DISCUSSION TITLE
React Native details

### DISCUSSION THREAD

**Discussion by SalvadorLekan** [TanStack/query#4273] [Q&A]
How do I get a component to listen to changes in data when the data has been re-fetched?
The only times the components know there's new data is if I changed the data with `client.setQueryData` or `client.resetQueries`.
This process is manual.
I want a way for the components to rerender when a query key cache has changed value.
How do I go about it?
PS: It's React Native.
Thank you.

**Comment by TkDodo** [comment]
> How do I get a component to listen to changes in data when the data has been re-fetched?

it should happen automatically. It's impossible to say what the problem is without seeing a reproduction.
